install.sh

```diff
@@ -129,46 +132,108 @@ check_deprecation() {
 # Function to dispatch to all legacy transformations
 #   at the start.
 #   We are not yet deleting  old values since the install
-#   could go wrong.
+#   could go wrong and dry-run could also be activated.
 #
-# Returns:
-#   1 when failed, 0 otherwise
+# Returns: 0 if succesful, 1 otherwise
 ############################################################
-legacy_transformations_start() {
+legacy_transform_before_update() {
+
+    LEGACY_TRANSFORM_FAILURES="false"
 
     # Variable transformations in global git config
     # Can be applied to all versions without any problem
     OLD_CONFIG_VALUE=$(git config --global githooks.autoupdate.updateCloneUrl)
     if [ -n "$OLD_CONFIG_VALUE" ]; then
-        git config --global githooks.cloneUrl "$OLD_CONFIG_VALUE"
+        git config --global githooks.cloneUrl "$OLD_CONFIG_VALUE" || LEGACY_TRANSFORM_FAILURES="true"
     fi
 
     OLD_CONFIG_VALUE=$(git config --global githooks.autoupdate.updateCloneBranch)
     if [ -n "$OLD_CONFIG_VALUE" ]; then
-        git config --global githooks.cloneBranch "$OLD_CONFIG_VALUE"
+        git config --global githooks.cloneBranch "$OLD_CONFIG_VALUE" || LEGACY_TRANSFORM_FAILURES="true"
     fi
 
     OLD_CONFIG_VALUE=$(git config --global githooks.previous.searchdir)
     if [ -n "$OLD_CONFIG_VALUE" ]; then
-        git config --global githooks.previousSearchDir "$OLD_CONFIG_VALUE"
+        git config --global githooks.previousSearchDir "$OLD_CONFIG_VALUE" || LEGACY_TRANSFORM_FAILURES="true"
     fi
 
     # Copy legacy file to new location
     if [ -f "$INSTALL_DIR/autoupdate/registered" ]; then
-        cp "$INSTALL_DIR/autoupdate/registered" "$INSTALL_DIR/registered"
+        cp "$INSTALL_DIR/autoupdate/registered" "$INSTALL_DIR/registered" || LEGACY_TRANSFORM_FAILURES="true"
+    fi
+
+    if [ "$LEGACY_TRANSFORM_FAILURES" = "true" ]; then
+        echo "! There were legacy transform errors: check stderr"
+        return 1
     fi
 
     return 0
 }
 
+############################################################
+# Tests if the commit sha `$1`
+#   is before or equal of the commit sha `$2`.
+#
+# Returns: 0 if succesful, 1 otherwise
+############################################################
+legacy_transform_is_ancestor() {
```

Comment:
By the way, does this say it's an ancestor once? (e.g. will the next update not say it's an ancestor of `ab86d2a529`?)
What if the update contains multiple commits, will it still work correctly?

Reply:
As the doc says:
```
legacy_transform_is_ancestor  "$INTERNAL_UPDATED_FROM_COMMIT" "ab86d2a529f58744a71e79227e434f19b84589e6"
```
checks if INTERNAL_UPDATED_FROM_COMMIT is <= ab86d2a529f58744a71e79227e434f19b84589e6 (which marks this PRs change on the master - 1commit, )
So that makes sense when you from now on : INTERNAL_UPDATED_FROM_COMMIT will be > ab8d

Reply:
You only needs transforms when you *hop* over this mark, makes sense?